### ISSUE TITLE
Paths with forward slashes in .editorconfig don't work on Windows

### ISSUE THREAD

**Issue by generalmimon** [editorconfig-checker/editorconfig-checker#281]
I use `ec-windows-amd64` (version [2.7.1](https://github.com/editorconfig-checker/editorconfig-checker/releases/tag/2.7.1)) and I wanted to have the following rule in .editorconfig:

```editorconfig
[bin/ksdump]
indent_size = 2
```

This is to check a file `ksdump` in a directory `bin`. Although I'm on Windows where historically the native path separator is the backslash `\` (but the forward slash `/` works just as well 99% of the time), the EditorConfig specification explicitly specifies that only forward slashes should be used in `.editorconfig` files (https://spec.editorconfig.org/#file-format):

> - Section Header: starts with a `[` and ends with a `]`.
> 
>   - Forward slashes (`/`) are used as path separators.
>   - Backslashes (`\\`) are **not allowed** as path separators (even on Windows).

Makes a lot of sense, because it would be very unwieldy if only `/` worked on Unix-style systems and only `\\` worked on Windows (for a cross-platform project, you'd need different variants of `.editorconfig`).

However, I found that on Windows, editorconfig-checker apparently doesn't find the `bin/ksdump` file if the path uses a forward slash (when I intentionally violate `.editorconfig` in the `bin/ksdump` file, `ec` doesn't report it):

```editorconfig
[bin/ksdump]
indent_size = 2
```

and it only finds the file when I use a backslash instead:

```editorconfig
[bin\\ksdump]
indent_size = 2
```

Which harms the idea of a cross-platform `.editorconfig` and goes against the EditorConfig spec.

**Comment by generalmimon** [comment]
In fact, not only the `bin/ksdump` file itself isn't found, but also when I use the brace syntax (`{s1,s2,s3}`) to match any of the given globs and include `bin/ksdump` as one of them, editorconfig-checker apparently doesn't apply the entire rule at all (it only starts working again when I remove `bin/ksdump`):

```console
$ cat .editorconfig
root = true

[*]
indent_style = space
trim_trailing_whitespace = true

[{*.rb,bin/ksdump,Gemfile}]
indent_size = 2

$ ec-windows-amd64.exe
README.md:
        25: Trailing whitespace

1 errors found

$ cat .editorconfig
root = true

[*]
indent_style = space
trim_trailing_whitespace = true

[{*.rb,Gemfile}]
indent_size = 2

$ ec-windows-amd64.exe
Gemfile:
        6: Wrong amount of left-padding spaces(want multiple of 2)
README.md:
        25: Trailing whitespace
lib/kaitai/struct/visualizer/node.rb:
        178: Wrong amount of left-padding spaces(want multiple of 2)
        179: Wrong amount of left-padding spaces(want multiple of 2)
        180: Wrong amount of left-padding spaces(want multiple of 2)
        181: Wrong amount of left-padding spaces(want multiple of 2)
        182: Wrong amount of left-padding spaces(want multiple of 2)
        183: Wrong amount of left-padding spaces(want multiple of 2)

8 errors found
```

Again, if I replace `bin/ksdump` with `bin\\ksdump`, it works fine.

**Comment by generalmimon** [comment]
After digging a bit deeper, this turns out to be a bug in https://github.com/editorconfig/editorconfig-core-go, so I'll close the issue here and report it there.
